jubilant/_version.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+import dataclasses
+import re
+from typing import Any
+
+# Per https://github.com/juju/juju/blob/afed87bbe98189805bc81e1af5d5fac70195fc86/core/semversion/version.go#L86-L91,
+# support version strings in the following formats:
+#
+# 1.2-release-arch
+# 1.2.3-release-arch
+# 1.2.3.4-release-arch
+# 1.2-alpha3-release-arch
+# 1.2-alpha3.4-release-arch
+majmin = r'^(\d+)\.(\d+)'
+relarch = r'([^-]+)-([^-]+)$'
+majmin_re = re.compile(rf'{majmin}-{relarch}')
+majmin_patch_re = re.compile(rf'{majmin}\.(\d+)-{relarch}')
+majmin_patch_build_re = re.compile(rf'{majmin}\.(\d+)\.(\d+)-{relarch}')
+majmin_tag_re = re.compile(rf'{majmin}-([a-z]+)(\d+)-{relarch}')
+majmin_tag_build_re = re.compile(rf'{majmin}-([a-z]+)(\d+)\.(\d+)-{relarch}')
+
+
+@dataclasses.dataclass(frozen=True)
+class Version:
+    """Parsed Juju CLI version as returned by ``juju version --format=json --all``.
+
+    The version from Juju is normally in the form ``major.minor.patch-release-arch``.
+    """
```

Comment:
Maybe it's better to add an example how to compare versions.

Apparently it's something like this but I had to read the source to figure that out.

```py
if juju.version().semver < (3, 6, 6): ...
```